hydroflow_macro/src/lib.rs

```diff
@@ -96,20 +98,28 @@ fn hydroflow_syntax_internal(
 /// Used for testing, users will want to use [`hydroflow_syntax!`] instead.
 #[proc_macro]
 pub fn hydroflow_parser(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let macro_invocation_path = proc_macro::Span::call_site().source_file().path();
+
     let input = parse_macro_input!(input as HfCode);
 
-    let flat_graph_builder = FlatGraphBuilder::from_hfcode(input);
-    let (flat_graph, _uses, diagnostics) = flat_graph_builder.build();
-    diagnostics.iter().for_each(Diagnostic::emit);
-    let flat_mermaid = flat_graph.mermaid_string_flat();
+    let flat_graph_builder = FlatGraphBuilder::from_hfcode(input, macro_invocation_path);
+    let (mut flat_graph, _uses, diagnostics) = flat_graph_builder.build();
+    if !diagnostics.iter().any(Diagnostic::is_error) {
+        flat_graph.merge_modules();
```

Comment:
.

Reply:
what does the dot mean?

Reply:
Oh, also could move `merge_modules` out of here if it is done automatically